barretenberg/cpp/src/barretenberg/vm2/common/memory_types.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+
+#include "barretenberg/vm2/common/field.hpp"
+
+namespace bb::avm2 {
+
+enum class MemoryTag {
+    FF,
+    U1,
+    U8,
+    U16,
+    U32,
+    U64,
+    U128,
+};
+
+using MemoryAddress = std::size_t;
```

Comment:
Not sure if this is the right choice to take size_t as it is a platform-specific type. It is the primitive type to address offsets in vector or array, but in our case if we use a map, we might well use uint32_t or explicitly uint64_t (with explicit memory bound checks)